plugin/src/lib/utilities/richTextConversion.ts

```diff
@@ -61,15 +82,31 @@ export const convertHtmlToLexical = (htmlString: string, editorConfig: Sanitized
       span: (el) => {
         const blockId = el && getAttributeValue(el, 'data-block-id')
         const blockType = el && getAttributeValue(el, 'data-block-type')
-        const translation = (blockTranslations?.['blocks'] || []).find((block: any) => block.id === blockId)
 
+        if (!blockId || !blockType) {
+          return undefined
+        }
+
+        if (blockType === 'pcsUpload') {
+          const relationTo = el && getAttributeValue(el, 'data-relation-to')
           return {
-          // use a relatively obscure name to reduce likelihood of a clash with an existing Slate editor configuration `nodeType`.
-          type: 'pcs-block',
-          // fieldName needed to obtain translations?
-          translation,
-          blockId,
-          blockType,
+            type: 'upload',
+            relationTo,
+            value: {
+              id: blockId,
+            },
+          }
```

Comment:
This structure represents how Slate upload nodes are stored in the database. A test is in place to ensure a similar Lexical node is created by the `convertSlateToLexical` function in `@payloadcms/richtext-lexical`.

Tests do not go as far as to check that this value gets populated with an appropriate relationship, this is beyond the scope.